src/mlpack/methods/range_search/rs_model_impl.hpp

```diff
@@ -487,6 +504,29 @@ inline bool& RSModel::Naive()
 {
   return boost::apply_visitor(NaiveVisitor(), rSearch);
 }
+//Save a RSModel into file
+inline void SaveModel(RSModel* model,std::string filename)
+{
+  std::ofstream ofs(filename);
+  boost::archive::text_oarchive oa(ofs);
+  oa << model;
+}
+//Load a RSModel from file
+inline std::string LoadModel(std::string filename)
+{
+  std::ifstream ifs(filename);
+  std::stringstream buffer;
+  buffer << ifs.rdbuf();
+  return buffer.str();
+}
+//Compare two RSModels by serialising them and then comparing their documents
+inline bool CheckModelSerial(RSModel* model1, RSModel* model2)
+{
+  std::string strmodel1="model1",strmodel2="model2";
+  SaveModel(model1,strmodel1);
+  SaveModel(model2,strmodel2);
+  return !LoadModel(strmodel1).compare(LoadModel(strmodel2));
+}
```

Comment:
I would suggest, instead of making this check function, to just make a function that returns a string.  Your `SaveModel()` function could be adapted:

```
std::string ModelToString(RSModel* model)
{
  std::ostringstream oss;
  boost::archive::text_oarchive oa(oss);
  oa << model;
  return oss.str();
}
```

Then you can compare two strings and make sure they are not the same. :+1: